### PR TITLE
LPS-47907 All users listed in Users page regardless of permissions when querying outside of any organization

### PR DESCRIPTION
**Pull request by pei-jung**
/cc @drewbrokke

### CODE REVIEW

**Comment by JorgeFerrer** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/apps/foundation/users-admin/users-admin-web/src/main/resources/META-INF/resources/view_flat_users.jsp [hunk 1]
```diff
@@ -266,7 +266,13 @@ boolean showRestoreButton = (searchTerms.getStatus() != WorkflowConstants.STATUS
 		}
 		else {
 			if (filterManageableOrganizations && !UserPermissionUtil.contains(permissionChecker, ResourceConstants.PRIMKEY_DNE, ActionKeys.VIEW)) {
-				userParams.put("usersOrgs", ArrayUtil.toLongArray(user.getOrganizationIds()));
+				long[] organizationIds = user.getOrganizationIds();
+
+				if (ArrayUtil.isEmpty(organizationIds)) {
+					organizationIds = new long[] {0};
```
Comment (JorgeFerrer) [inline modules/apps/foundation/users-admin/users-admin-web/src/main/resources/META-INF/resources/view_flat_users.jsp]: Hey Pei-Jung,

Why is this needed? Is it because organizationIds is null? 
If so, shouldn't we change the implementation of user.getOrganizationIds() instead so that such a thing never happens.

Comment (pei-jung) [inline modules/apps/foundation/users-admin/users-admin-web/src/main/resources/META-INF/resources/view_flat_users.jsp]: Hey Jorge,

user.getOrganizationIds() returns an empty string if the user does not belong to any organization, in which i think the implementation is correct. The problem is when using the `usersOrgs` param, it does a joint/filter on the users_orgs table on the search results. If the `usersOrgs` param has an empty array, the results wont be filtered based on the organizations. Because of that, a user without any organization is able to view all the users. We need to add a default organization id when organizationIds is empty to ensure that the results is filtered correctly.

File: modules/apps/foundation/users-admin/users-admin-web/src/main/resources/META-INF/resources/view_flat_users.jsp [hunk 2]
```diff
@@ -266,7 +266,13 @@ boolean showRestoreButton = (searchTerms.getStatus() != WorkflowConstants.STATUS
 		}
 		else {
 			if (filterManageableOrganizations && !UserPermissionUtil.contains(permissionChecker, ResourceConstants.PRIMKEY_DNE, ActionKeys.VIEW)) {
```
Comment (pei-jung) [inline modules/apps/foundation/users-admin/users-admin-web/src/main/resources/META-INF/resources/view_flat_users.jsp]: this condition checks the permission. if the user doesnt have permission to view all users in the portal, only shows users that belong to the organzations that the user manages.

File: modules/apps/foundation/users-admin/users-admin-web/src/main/resources/META-INF/resources/view_flat_users.jsp (hunk 1 above)
Comment (JorgeFerrer) [inline modules/apps/foundation/users-admin/users-admin-web/src/main/resources/META-INF/resources/view_flat_users.jsp]: Hey Pei-Jung,

Thanks for the explanation, it makes sense.

However I'm not seeing that in the code. In line 272 you are still initializing organizationIds to an empty array. Can you clarify that?

Comment (pei-jung) [inline modules/apps/foundation/users-admin/users-admin-web/src/main/resources/META-INF/resources/view_flat_users.jsp]: Hey Jorge,

In line 272, it's not an empty array. It's an array with the default organization id 0: `{0}`.

Comment (JorgeFerrer) [inline modules/apps/foundation/users-admin/users-admin-web/src/main/resources/META-INF/resources/view_flat_users.jsp]: You are right, sorry, I misread it.

However I must say that now it seems even more strange. I mean it's like a magic value (an array with one element which is zero) which creates further questions such as "Does the value of the array have to be 0? Could it be other value?". This is subject to be misunderstood by future developers (including our future selves) who could introduce bugs as a result.

To avoid this, I think we might want to introduce a constant which represents NO_ORGANIZATIONS and pass it over. The same constant should then be used in the finder to represent that value.

What do you think?

Comment (pei-jung) [inline modules/apps/foundation/users-admin/users-admin-web/src/main/resources/META-INF/resources/view_flat_users.jsp]: Hey Jorge,

I agree with you. Having a constant will definitely make the logic more clear and avoid any confusion in the future. I will make the changes and resend you a pull. Thanks.
